parse_tz.c

```diff
@@ -665,7 +665,7 @@ timelib_time_offset *timelib_get_time_zone_info(timelib_sll ts, timelib_tzinfo *
 	int32_t offset = 0, leap_secs = 0;
 	char *abbr;
 	timelib_time_offset *tmp = timelib_time_offset_ctor();
-	timelib_sll                transition_time;
+	timelib_sll                transition_time = 0;
```

Comment:
I don't think that this is the right fix. You're right in saying that ``tmp->transition_time`` can be uninitialised when ``tmp`` is returned from the function, but just initialising it with ``0`` is not correct. The assumption should be that the if ``fetch_timezone_offset`` returns non-NULL, that ``transition_time`` is set.

I recently added the three lines at [618-620](https://github.com/derickr/timelib/pull/65/files#diff-24275a3aeca7004ac75f13d604b83e41R618-R620), where not-null is returned from ``fetch_timezone_offset``, but ``transition_time`` is not set. Instead of returning ``0``, it should set the value to ``INT64_MIN`` (and also in line [607](https://github.com/derickr/timelib/pull/65/files#diff-24275a3aeca7004ac75f13d604b83e41R607)).

Reply:
I've made the suggested change to return `INT64_MIN` in case `transition_time` is not set.

Any half-decent compiler should find this. There must not exist any code path where uninitialized variables are used, regardless of the input. I'm not aware of a current situation (input) where this causes a problem, but this does not exclude the possibility that such a situation (input) exists.